### ISSUE TITLE
Need to be able to pre-build library for wasm compilation from rust

### ISSUE THREAD

**Issue by Arcahub** [nrwl/nx#9890]
Hello,

## Description
I would like to be able to pre-build library / application in the workspace for development server.

## Motivation

I'm using @nrwl/nx to manage a monorepo with a Nest app and a rust lib that will be build in wasm.

I'm using the community plugin `@nxrs/cargo` to integrate the cargo workspace, no problem here.

I've added a build target to the rust library to build it in wasm using wasm-pack.

I've added the wasm lib result in my tsconfig paths.

And finally I'm importing the wasm lib in my Next app.

At this point when I run `yarn nx build my-next-app` everything is working fine. My rust library is built as a wasm library and the build result use it well.

The problem came when I want to start a development server using `yarn nx serve my-next-app`. 

Nx seems to read directly library code without building them but in my case there is no source code to read and you need to build the library. 

I've a test repository with a branch named `feat/nx-issue`: https://github.com/Arcahub/rust-wasm-next-nx/tree/feat/nx-issue

You can clone the repository, checkout the branch feat/nx-issue. The application is located in `apps/my-next-app` and the rust library is in `libs/my-rust-lib`.

You can try to start the development server with `yarn nx serve my-next-app`, you should have an error like this one:

```
Module not found: Can't resolve '@rust-wasm-next/my-rust-lib'
   8 |   loader: async () => {
   9 |     // Import the wasm module
> 10 |     const module = await import('@rust-wasm-next/my-rust-lib');
     |                         ^
  11 | 
  12 |     // Return a React component that calls the add_one method on the wasm module
  13 |     return React.memo(

https://nextjs.org/docs/messages/module-not-found
```

You can then build the wasm lib from rust with `yarn nx build my-rust-lib` and retry to start the development server and every things should be working. 

Notes: Remember to clean the `dist` folder and the cache in `node_modules/.cache` between try to work in a clean environment

## Suggested Implementation
My first naive thought was to work on the task graph of nx but I only found capacity to work on the dependency graph and not the task graph. Maybe it would also be a good feature to have in nx if it doesn't already exist.

## Alternate Implementations
A custom tag in the project.json of the library that enable pre-build ? And also if watch mode can be enable for this type of build to rebuild them when source code change. In my case when I update the rust code, it would be incredible to have the wasm library automatically rebuild.

Thanks for reading, if you have any question feel free to ask me.

PS: The name of the issue might not be good you can change it.

**Comment by AgentEnder** [comment]
@Arcahub `watch` would not work for the rust code in this case, but if you update your next code's serve target with:
```jsonc
{
  // ... other project configuration
  "targets": {
    "serve": {
      // ... existing configuration
      "dependsOn": {
        "target": "build-wasm",
        "projects": "dependencies"
      }
    }
  }
}
```

This assumes that the build target for your rust wasm project is called build-wasm, you could specify it as "build" as well, but Nx may pick up other prebuild steps as well in that case.

**Comment by Arcahub** [comment]
Thanks for your answer. 
If I understand well the config make nx run the target build-wasm for every dependency of my project that has this target ?

I will try that later today and close this issue if it helps or sends you feedback otherwise.

**Comment by Arcahub** [comment]
I just tested and it's working fine as expected. Thanks you very much :1st_place_medal: .

I tested to start the development server a first time, stop it, make a modification to the rust lib and restart the development server and the lib is well updated so it's really nice. Only sad point is that as you said it in your post I don't have a hot reload / watch mode but I might be able to hack it with a custom target using the same method.

Again thanks you I will close this issue since my problem is resolved.

PS: I updated my test repository so people can test it.
